docling_core/utils/legacy.py

```diff
@@ -342,5 +349,276 @@ def _make_spans(cell: TableCell, table_item: TableItem):
+                                else None
+                            )
+                            cell = TableCell(
+                                start_row_offset_idx=row_ix,
+                                end_row_offset_idx=row_ix + 1,
+                                start_col_offset_idx=col_ix,
+                                end_col_offset_idx=col_ix + 1,
+                                text=orig_cell_data.text,
+                                bbox=cell_bbox,
+                                column_header=(orig_cell_data.obj_type == "col_header"),
+                                row_header=(orig_cell_data.obj_type == "row_header"),
+                                row_section=(orig_cell_data.obj_type == "row_section"),
+                            )
+
+                            if orig_cell_data.spans is not None:
+                                # convert to a tuple of tuples for hashing
+                                spans_tuple = tuple(
+                                    tuple(span) for span in orig_cell_data.spans
+                                )
+
+                                # skip repeated spans
+                                if spans_tuple in seen_spans:
+                                    continue
+
+                                seen_spans.add(spans_tuple)
+
+                                cell.start_row_offset_idx = min(
+                                    s[0] for s in spans_tuple
+                                )
+                                cell.end_row_offset_idx = (
+                                    max(s[0] for s in spans_tuple) + 1
+                                )
+                                cell.start_col_offset_idx = min(
+                                    s[1] for s in spans_tuple
+                                )
+                                cell.end_col_offset_idx = (
+                                    max(s[1] for s in spans_tuple) + 1
+                                )
+
+                                cell.row_span = (
+                                    cell.end_row_offset_idx - cell.start_row_offset_idx
+                                )
+                                cell.col_span = (
+                                    cell.end_col_offset_idx - cell.start_col_offset_idx
+                                )
+
+                            table_data.table_cells.append(cell)
+
+                new_item = doc.add_table(data=table_data, prov=prov)
+                if (caption_item := floating_to_caption.get(ix)) is not None:
+                    if caption_item.text is not None:
+                        caption_prov = _transform_prov(caption_item)
+                        caption = doc.add_text(
+                            label=DocItemLabel.CAPTION,
+                            text=caption_item.text,
+                            prov=caption_prov,
+                            parent=new_item,
```

Comment:
Just to keep it in mind: this is currently not allowed according to the data schema (`parent` can only be `None` or a `GroupItem` but not a `TableItem`). With PR #103 , the constraint would be loosened to just a `NodeItem`.